src/pytorch_lightning/trainer/trainer.py

```diff
@@ -388,6 +389,9 @@ def __init__(
                 and smaller datasets reload when running out of their data. In 'min_size' mode, all the datasets
                 reload when reaching the minimum length of datasets.
                 Default: ``"max_size_cycle"``.
+
+            inference_grad_mode: Controls if autograd is used in the inference phase, such as during model testing.
```

Comment:
default should be True

Reply:
setting `inference_grad_mode=True` as the default causes `def test_evaluation_move_metrics_to_cpu_and_outputs(tmpdir, accelerator):` to fail when combined with:     
```
if not inference_grad_mode:
        context_manager_class = torch.no_grad
```